src/WebJobs.Script.Grpc/Extensions/ScriptInvocationContextExtensions.cs

```diff
@@ -149,6 +156,33 @@ private static bool ShouldSkipBindingData(KeyValuePair<string, object> bindingDa
             return false;
         }
 
+        /// <summary>
+        /// Checks if the object being passed is converted into a form that can be sent to the worker.
+        /// This is required in cases where an object conversion was being delayed for the <see cref="SharedMemoryManager"/> to handle,
+        /// but for some reason (error, out of memory etc.) it was unable to do so.
+        /// Therefore, we convert it to a form that can be sent without requiring shared memory.
+        /// </summary>
+        /// <param name="inputVal">Object to check if it is in a valid form.</param>
+        /// <param name="convertedVal">Output object in a valid form if the input was invalid, <see cref="null"/> otherwise.</param>
+        /// <returns><see cref="true"/> if a conversion took place, <see cref="false"/> otherwise.</returns>
+        private static bool TryConvertObjectIfNeeded(object inputVal, out object convertedVal)
+        {
+            convertedVal = null;
+
+            if (inputVal is CacheableObjectStream stream)
+            {
+                using (MemoryStream ms = new MemoryStream())
+                {
+                    stream.CopyTo(ms);
```

Comment:
This was a double operation right? First copying to memory and then copy to an array.
Did we have a more efficient way of just copying to the final array?